packages/rainbowkit/src/wallets/walletConnectors/okxWallet/okxWallet.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import { InjectedConnector } from 'wagmi/connectors/injected';
+import { Chain } from '../../../components/RainbowKitProvider/RainbowKitChainContext';
+import { Wallet } from '../../Wallet';
+import { getWalletConnectConnector } from '../../getWalletConnectConnector';
+
+export interface OKXWalletOptions {
+  chains: Chain[];
+}
+
+function isOKX(ethereum: NonNullable<typeof window['ethereum']>) {
+  // `isOkxWallet` or `isOKExWallet` needs to be added to the wagmi `Ethereum` object
+  // @ts-expect-error
+  const isOKX = Boolean(ethereum.isOkxWallet || ethereum.isOKExWallet);
+  if (!isOKX) {
+    return false;
+  }
+
+  return true;
+}
+
+export const okxWallet = ({ chains }: OKXWalletOptions): Wallet => {
+  const isOKXInjected =
+    typeof window !== 'undefined' &&
+    typeof window.ethereum !== 'undefined' &&
+    isOKX(window.ethereum);
+
+  const shouldUseWalletConnect = !isOKXInjected;
+
+  return {
+    id: 'okx',
+    name: 'OKX Wallet',
+    iconUrl: async () => (await import('./okxWallet.svg')).default,
+    iconAccent: '#000',
+    iconBackground: '#000',
+    downloadUrls: {
+      browserExtension:
+        'https://chrome.google.com/webstore/detail/okx-wallet/mcohilncbfahbmgdjkbpemcciiolgcge',
+      android: 'https://static.byucvwj.cn/upgradeapp/okx-android.apk',
+      ios: 'https://itunes.apple.com/app/id1327268470?mt=8',
+      qrCode: 'https://www.okx.com/web3',
+    },
+    createConnector: () => {
+      const connector = shouldUseWalletConnect
+        ? getWalletConnectConnector({ chains })
+        : new InjectedConnector({
+            chains,
+          });
+
+      const getUri = async () => (await connector.getProvider()).connector.uri;
```

Comment:
On iOS this URI will default to `wc://` and iOS selects a wallet based on the user's usage pattern. Does OKX have a Deep Link or Universal Link for this pairing flow? We will need to be explicit like so:
```
const getUri = async () => {
   const { uri } = (await connector.getProvider()).connector;
   return isAndroid()
     ? uri
     : `https://rnbwapp.com/wc?uri=${encodeURIComponent(uri)}`;
};
```